web/srv/api_handlers.go

```diff
@@ -441,6 +441,28 @@ func (h *handler) handleAPIResourceDefinition(w http.ResponseWriter, req *http.R
 	w.Write(resourceDefinition)
 }
 
+func (h *handler) handleGetExtensions(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	ctx := req.Context()
+	linkerdExtensionLabel := "linkerd.io/extension"
+	nsList, err := h.k8sAPI.CoreV1().Namespaces().List(ctx, metav1.ListOptions{
+		LabelSelector: linkerdExtensionLabel,
+	})
+	if err != nil {
+		renderJSONError(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	resp := []map[string]interface{}{}
+	for _, ns := range nsList.Items {
+		resp = append(resp, map[string]interface{}{
+			"extensionName": ns.GetLabels()[linkerdExtensionLabel],
+			"namespace":     ns.Name,
+		})
+	}
+
+	renderJSON(w, resp)
+}
```

Comment:
So in `pkg/k8s` we already have a `GetNamespaceWithExtensionLabel` which will get the namespace for a given label value. Since this check only exists for the multicluster extension, we can use that.

We'd still need the handling here for when that method returns an error, but I don't think we need to get all namespaces and loop through them when we already know the label value we are looking for.

Reply:
That's a good point -- I think it'd be still useful to get a list of all the extensions for the frontend in case we'd like to check if other extensions apart from multicluster exist. I could change this method so that the frontend can pass the extension name in the request. If the extension exists, we will return it in the response, if it doesn't we can send a not found or an empty object. WDYT @kleimkuhler ?

Reply:
Yea passing the extension name is what I'd recommend so we can use `GetNamespaceWithExtensionLabel`. Since we don't need to get any other extensions, I think solving for this case is okay. If we come to a point where we need to get lists of extensions in the future, then maybe this can change then.